Got rid of old spring-social dependency

MarekB01 opened the pull request:
Related to: https://github.com/BroadleafCommerce/QA/issues/4974

Removed spring-social dependency as it became deprecated. Rewrote BroadleafSocialRegisterController to use OAuth2 implementation.

MarekB01 committed:
BroadleafCommerce/QA-4974_get_rid_of_old_spring-social_dependency_6_3

Removed spring-social dependency as it became deprecated. Rewrote BroadleafSocialRegisterController to use OAuth2 implementation.

riteshadhikari17 commented:
This is the error in jenkins
```
[ERROR]     'dependencies.dependency.version' for org.springframework.security:spring-security-oauth2-client:jar is missing. @ line 101, column 21
```

FYI Module Parent is for modules. Core does not use MP. Define the version here and close the PR for MP.

MarekB01 committed:
BroadleafCommerce/QA-5038_Update_GA4_processor

Refactored code, renamed properties to Google analytics 4. Added spring-security-oauth2-client dependency in core module

MarekB01 commented:
Now it can be merged

riteshadhikari17 commented on core/broadleaf-profile-web/pom.xml:
```diff
@@ -81,5 +81,11 @@
+        <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-oauth2-client</artifactId>
+            <version>5.5.3</version>
```
Don't define version here. It should come from root pom.

MarekB01 committed:
BroadleafCommerce/QA-5038_Update_GA4_processor

Refactored code, renamed properties to Google analytics 4. Added spring-security-oauth2-client dependency in core module. Refactored code

riteshadhikari17 commented on pom.xml:
```diff
@@ -1162,6 +1162,11 @@
+            <dependency>
+                <groupId>org.springframework.security</groupId>
+                <artifactId>spring-security-oauth2-client</artifactId>
+                <version>6.0.2</version>
```
This version needs to sync with spring security. We define it as property above so use it.
`${spring.security.version}`. Also shouldn't you remove the spring-social dependency below?

oleksiimiroshnyk commented on core/broadleaf-framework-web/pom.xml:
```diff
@@ -98,5 +98,9 @@
             <groupId>org.easymock</groupId>
             <artifactId>easymockclassextension</artifactId>
         </dependency>
+        <dependency>
```
I see you've added a new dependency, but what about deleting old spring social ? Is it still there ?

MarekB01 committed:
BroadleafCommerce/QA-4974_get_rid_of_old_spring-social_dependency_6_3

Removed spring-social dependency as it became deprecated. Rewrote BroadleafSocialRegisterController to use OAuth2 implementation.

riteshadhikari17 committed:
Merge branch 'develop-7.0.x' into QA-4974_get_rid_of_old_spring-social_dependency_6_3
